app/util/session_id.py

```diff
@@ -2,7 +2,9 @@
 
 
 class SessionId(object):
+    EXPECTED_SESSION_HEADER_KEY = 'Expected-Session-Id'
```

Comment:
I added this as a way to more clearly distinguish between the request header and the response header. Previously a "Session-Id" request header was set to indicate the *expected* session id, and a "Session-Id" response header was set to communicate what was the actual current session id.

It's not too bad but I was thinking using a different header name would be clearer. This is also closer to the pattern followed by some standard headers such as "Accept" and "Content-Type" -- the first one is something you'd put in a request to declare what types you accept, and the second is what you'd find in the response to declare what type it is.